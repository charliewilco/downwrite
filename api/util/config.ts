const address =
  process.env.NODE_ENV === "production"
    ? `${process.env.DB_USER}:${process.env.DB_PASSWORD}@${process.env.DB_ADDRESS}`
    : `127.0.0.1:27017/downwrite`;

const Config = {
  key: process.env.SECRET_KEY || "1a9876c4-6642-4b83-838a-9e84ee00646a",
  dbCreds: `mongodb://${process.env.MONGO_URL || address}`
};

export default Config;
