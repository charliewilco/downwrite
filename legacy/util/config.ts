const Config = {
  key: process.env.SECRET_KEY || "1a9876c4-6642-4b83-838a-9e84ee00646a",
  dbCreds: `mongodb://${process.env.CURRENT_DB_ADDRESS}`
};

export default Config;
