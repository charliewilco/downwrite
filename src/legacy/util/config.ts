const Config = {
  key: process.env.SECRET_KEY || "1a9876c4-6642-4b83-838a-9e84ee00646a",
  dbCreds: process.env.ATLAS_DB_ADDRESS,
};

export default Config;
