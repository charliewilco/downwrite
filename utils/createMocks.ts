import uuid from "uuid/v4";
import * as Draft from "draft-js";
import { IPost } from "downwrite";

export const createMockPost = (overides?: Partial<IPost>): IPost => {
  const post = Draft.EditorState.createWithContent(
    Draft.ContentState.createFromText("Hello")
  );
  const content = Draft.convertToRaw(post.getCurrentContent());
  const dateAdded = "2019-04-14T07:40:08.591Z";
  const id = uuid();

  return Object.assign(
    {},
    {
      title: "Something",
      id,
      dateAdded,
      content,
      public: false
    },
    overides || {}
  );
};

export const createMockPosts = (count: number): IPost[] => {
  return Array(count || 1)
    .fill(null)
    .map(() => createMockPost({ title: "Mocked Posts" }));
};

export const createEditorState = (content: string): Draft.EditorState => {
  const state = Draft.EditorState.createWithContent(
    Draft.ContentState.createFromText("Hello")
  );

  return state;
};
