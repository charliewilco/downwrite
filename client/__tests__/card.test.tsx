import Card from "../components/card";
import { fireEvent, render } from "react-testing-library";
import { createMockPost } from "./config/createMocks";

const mockDelete = jest.fn();

jest.mock("next/link", () => {
  return ({ children }: { children: React.ReactNode }) => {
    return children;
  };
});

const post = createMockPost({ title: "Starting Again" });

const PostedCard: React.SFC<any> = () => (
  <Card public={false} {...post} onDelete={mockDelete} />
);

const { container, getByTestId } = render(<PostedCard />);

describe("<Card />", () => {
  it("contains snippet from content", () => {
    const snippet = getByTestId("CARD_EXCERPT").textContent;
    expect(snippet.length).toBeLessThanOrEqual(90);
  });

  it("contains a title", () => {
    expect(post.title).toBe("Starting again");
    expect(getByTestId("CARD_TITLE")).toHaveTextContent("Starting again");
  });

  it("contains delete button", () => {
    expect(getByTestId("CARD_DELETE_BUTTON")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should fire a delete", () => {
    fireEvent.click(getByTestId("CARD_DELETE_BUTTON"));

    expect(mockDelete).toHaveBeenCalled();
  });
});
