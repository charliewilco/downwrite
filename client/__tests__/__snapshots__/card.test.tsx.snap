// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> matches snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 192px;
  font-weight: 400;
  box-shadow: 0 0 2px rgba(0,0,0,0.07),0 2px 4px rgba(0,0,0,0.12);
}

.c0 a {
  cursor: pointer;
}

.c2 {
  font-size: 14px;
  margin-bottom: 0px;
  font-weight: 700;
}

.c1 {
  display: block;
  border-bottom: 1px solid;
  padding: 12px 8px;
}

.c4 {
  font-size: 12px;
  display: block;
  font-weight: 400;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px;
  font-weight: 700;
  font-size: 12px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 a:hover {
  color: #4382A1;
}

.c6 {
  font-size: 12px;
}

.c5 {
  display: block;
  padding: 0.5rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
}

.c8 {
  border: 0px;
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-family: inherit;
  box-sizing: inherit;
  font-size: 12px;
  -webkit-font-smoothing: antialiased;
}

.c3 {
  margin-right: 8px;
}

@media (min-width:57.75rem) {
  .c2 {
    font-size: 18px;
  }
}

<div
  class="c0"
  data-testid="CARD"
>
  <header
    class="c1"
  >
    <h2
      class="c2"
      data-testid="CARD_TITLE"
    >
      <a
        class="c3"
      >
        Starting again
      </a>
    </h2>
    <small
      class="c4"
    >
      added 
      about 1 year
       ago
    </small>
  </header>
  <div
    class="c5"
  >
    <p
      class="c6"
      data-testid="snippet"
    >
      What am i really going to do here?
    </p>
  </div>
  <footer
    class="c7"
  >
    <div
      data-testid="CARD_EXCERPT"
    >
      <a
        class="c3"
      >
        Edit
      </a>
    </div>
    <button
      class="c8"
      data-testid="CARD_DELETE_BUTTON"
    >
      Delete
    </button>
  </footer>
</div>
`;
